source/common/access_log/access_log_formatter.cc

```diff
@@ -110,13 +110,19 @@ std::vector<FormatterPtr> AccessLogFormatParser::parse(const std::string& format
         current_token = "";
       }
 
-      size_t command_end_position = format.find('%', pos + 1);
-      if (command_end_position == std::string::npos) {
-        throw EnvoyException(fmt::format(
-            "Incorrect configuration: {}. Expected end of operation '%', around position {}",
-            format, pos));
+      static const std::regex command_w_args_regex("%([A-Z]|_)+(\\([^\\)]*\\))?(:[0-9]+)?(%)");
```

Comment:
Static non-PoD are disallowed due to static initialization fiasco in C++. See https://github.com/envoyproxy/envoy/blob/master/STYLE.md for suggested workarounds.

Reply:
TIL'd. Looking into

Reply:
since the access log formats are only parsed on start-up, theoretically once (in actuality twice, once for default format and once for actual format, which is something I might follow-up on...) I think it would be fine to just declare this as a `const std::regex` prior to the for-loop. After starting, Envoy doesn't re-parse access log formats so making this `static` doesn't yield any actual gain. The [alternative idiom](https://github.com/envoyproxy/envoy/blob/master/include/envoy/registry/registry.h#L83) seems heavyweight for the use case.